src/utils.ts

```diff
@@ -296,6 +306,55 @@ const useEnveloping = (
   return false;
 };
 
+const maxPossibleGasVerification = async (
+  transaction: PopulatedTransaction,
+  gasPrice: BigNumberish,
+  gasLimit: BigNumberish,
+  workerAddress: string
+): Promise<{ maxPossibleGas: BigNumber; transactionResult: string }> => {
+  let maxPossibleGasAttempt = BigNumber.from(gasLimit.toString());
+  const provider = getProvider();
+
+  for (let i = 1; i <= GAS_VERIFICATION_ATTEMPTS; i++) {
+    try {
+      log.debug(
+        `attempting with gasLimit : ${maxPossibleGasAttempt.toString()}`
+      );
+
+      const transactionResult = await provider.call(
+        {
+          ...transaction,
+          from: workerAddress,
+          gasPrice,
+          gasLimit: maxPossibleGasAttempt,
+        },
+        'pending'
+      );
+
+      return { maxPossibleGas: maxPossibleGasAttempt, transactionResult };
+    } catch (e) {
+      const error = e as Error;
+      if (!error.message.includes('Not enough gas left')) {
+        throw error;
+      }
+      maxPossibleGasAttempt = applyFactor(gasLimit, FACTOR * i);
+    }
+  }
```

Comment:
I think that this logic is different from what we had before. Before the factory was increased by doubling it for each iteration, while now it grows linearly.
Furthermore, with this loop, the first two iterations we try the same value, don't we?

Reply:
I can move it back to the previous logic but it accomplish almost the same, and I think that is simpler, this does not affect the logic.  
previous version:
.25*2=.5
.5*2=1
1*2=2

new version:
.25*1=.25
.25*2=.5
.25*3=.75
.25*4=1

Reply:
updated